From e90d502faafab49e039a2e621e693d5076577f4f Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Tue, 3 Feb 2026 17:24:03 -0600
Subject: [PATCH 187/268] wined3d: Clear backup_dc and backup_wnd when deleting
 them.

If we were using a backup DC and then perform a reset, we will clear the DC, but
not zero it, meaning that once 3D is initialized again, we may eventually end up
trying to use the same backup DC which has been deleted. Fix this by also
zeroing the DC pointer.
---
 dlls/wined3d/device.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/wined3d/device.c b/dlls/wined3d/device.c
index 5956d672ddf..b87c670463e 100644
--- a/dlls/wined3d/device.c
+++ b/dlls/wined3d/device.c
@@ -1487,6 +1487,8 @@ void wined3d_device_gl_delete_opengl_contexts_cs(void *object)
 
         wined3d_release_dc(device_gl->backup_wnd, device_gl->backup_dc);
         DestroyWindow(device_gl->backup_wnd);
+        device_gl->backup_dc = NULL;
+        device_gl->backup_wnd = NULL;
     }
 }
 
-- 
2.55.0

